source/MaterialXCore/Element.cpp

```diff
@@ -81,6 +81,84 @@ bool Element::operator!=(const Element& rhs) const
     return !(*this == rhs);
 }
 
+bool Element::isEquivalent(ConstElementPtr rhs, ElementEquivalenceOptions& options,
+                           ElementEquivalenceResult* result) const
+{
+    if (getCategory() != rhs->getCategory() ||
+        getName() != rhs->getName())
+    {
+        if (result)
+            result->addMessage("Name or category does not match for elements: '" + getNamePath() + "', '" + rhs->getNamePath() + "'");
+        return false;
+    }
+
+    // Compare attribute names. 
+    StringVec attributeNames = getAttributeNames();
+    StringVec rhsAttributeNames = rhs->getAttributeNames();
+
+    // Filter out any attributes specified in the options.
+    const StringSet& skipAttributes = options.skipAttributes;
+    if (!skipAttributes.empty())
+    {
+        attributeNames.erase(std::remove_if(attributeNames.begin(), attributeNames.end(),
+            [&skipAttributes](const string& attr) { return skipAttributes.find(attr) != skipAttributes.end(); }),
+            attributeNames.end());
+        rhsAttributeNames.erase(std::remove_if(rhsAttributeNames.begin(), rhsAttributeNames.end(),
+            [&skipAttributes](const string& attr) { return skipAttributes.find(attr) != skipAttributes.end(); }),
+            rhsAttributeNames.end());
+    }    
+
+    // Ignore ordering if option specified.
+    if (options.ignoreAttributeOrder)
```

Comment:
Since attribute order has no _functional_ effect on a document, I think it's reasonable to simplify this method and assume that attribute order is ignored when testing equivalence.

Reply:
I mostly added this in as @ld-kerley mentioned it would be good to support both syntactical as well as functional logic. I'm fine with removing it.

Reply:
@jstone-lucasfilm - do we know that no application that renders a UI for MaterialX relies on the order of the attributes defined in the document?  

If we are taking the opinion that attribute order on an element is irrelevant, that should probably be explicitly stated in the specification somewhere, if it isn't already (I didn't check).

Reply:
I'd thought that the goal of `isEquivalent` was to test functional, rather than syntactical equivalence.  After all, we already have an equality test that is designed to test syntactical equivalence!

For functional/rendering equivalence, there's no difference produced when reordering child elements or reordering attributes, but of course both operations affect syntactical equivalence.

Reply:
To @ld-kerley's point, any syntactical change can affect the display of a MaterialX document in an application UI, even the introduction of a space in a value string or the swapping of the order of two elements in the document.  Neither of these can affect rendering, however, so it's important that we choose which type of equivalence we're testing here.

Reply:
I'm perfectly fine with make this a functional test only and removing unneeded options. `operator==` already handles all syntactical differences. Thus it's clear that one interface is purely for functional and one is purely for syntactical.

For syntactical differences affecting display, can we spawn a new thread -- where I'd suggest looking at *explicit* display meta-data  to address ambiguity for things like implied syntactical order.  Where possible, the documentation should denote where ordering matters (e.g. node input / output signature ordering) and where it may affect display (attribute characteristics).

Reply:
That sounds like a good approach to me, @kwokcb.

For document display in an application UI, I would note that virtually all syntactic changes to a document will affect its display for _some_ application, even the most trivial change to a value string, so a differentiation between exact document equality and UI equality is not likely to be universal or valuable.

For functional/render equivalence, I would say that this is what we're defining in your proposed isEquivalent method, and documenting these same rules in a future version of the MaterialX specification makes good sense.